apps/reference/docs/guides/platform/logs.mdx

```diff
@@ -4,78 +4,177 @@ title: Logging
-:::
+```sql
+SELECT timestamp, r.method, h.cf_ipcountry
+FROM edge_logs t
+CROSS JOIN UNNEST(t.metadata) as m
+CROSS JOIN UNNEST(m.request) as r
+CROSS JOIN UNNEST(r.headers) as h
+```
 
-### Functions
+Here, the `metadata` key when rendered in the Logs Explorer shows an array. In order to drill down further, we perform the `CROSS JOIN UNNEST()` to select the nested key `request`. However, in order to query the headers, we have to go one step further and perform the join and unnest again in order to have access to the headers, under the alias `h`.
 
-You may have also noticed from the above examples that we are able to use certain SQL functions within our queries. The querying engine currently supported is BigQuery, hence you may use [any valid BigQuery function](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators) within your query.
+### LIMIT and Result Row Limitations
 
-#### Timestamp Behavior
+The Logs Explorer has a maximum of 1000 rows per run. To optimize your queries, you may wish to add in a `LIMIT` to your SQL query, to reduce the number of rows returned even further.
 
-Each log entry is stored with a `timestamp`. In order to utilize the `timestamp` field in a query, you can use the appropriate [timestamp functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp).
+### Best Practices
 
-:::note
+1. Have a WHERE statement filtering over **timestamp**
 
-In the Log Explorer, timestamps are rendered as unix microsecond timestamps. SQL queries, however, should always use the `TIMESTAMP` data type. If you are using a unix timestamp value in a query, cast the value to a `TIMESTAMP` data type.
+  Querying across your entire log retention range might seem appealing, especially when you just need to get your query results fast. However, for **Enterprise** customers that have a large retention range, you would run the risk of your queries timing out due to the length cruch time required to scan and analyze across your large dataset. We recommend optimizing queries based on timestamp filtering, to ensure
 
-:::
 
-## Templates
+2. Avoid selecting large nested objects. Select individual values instead.
+
+  When querying large object, the columnar storage engine (in this case BigQuery) will select each column associated with each nested key, resulting in a large number of columns being selected. This inadvertently impacts the query speed and may result in timeouts or memory errors, especially for projects with large amount of logging activity. 
+
+  Instead, try to select only the values required, and where this is not possible, drill down to the minimally required object keys.
+
+  ```sql
+  -- ❌ edge_logs metadata contains many nested keys
+  SELECT timestamp, m as metadata
+  FROM edge_logs t
+  CROSS JOIN UNNEST(t.metadata) as m;
```

Comment:
Thoughts about commenting out the actual statement too so it's more apparent as something to not do?

Reply:
Hmm, the syntax highlighting would make it more readable though 🤔 
I've shifted the comment on the nested keys to be inline, and added a more obvious "Avoid doing this" comment